protocols/grpc/utils.go

```diff
@@ -287,3 +299,86 @@ func ResourcesGRPCtoOCI(grpcResources *LinuxResources) (*specs.LinuxResources, e
 
 	return s, err
 }
+
+// ChangeToBundlePath changes the cwd to the bundle path defined in the OCI spec
+func ChangeToBundlePath(spec *specs.Spec) (string, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return cwd, err
+	}
+
+	if spec == nil || spec.Root == nil || spec.Root.Path == "" {
+		return cwd, errors.New("Invalid OCI spec")
+	}
+
+	rootfsPath := spec.Root.Path
+	if !filepath.IsAbs(rootfsPath) {
+		rootfsPath = filepath.Join(cwd, rootfsPath)
+	}
+
+	bundlePath := filepath.Dir(rootfsPath)
+
+	err = os.Chdir(bundlePath)
+	return cwd, err
+}
+
+// WriteSpecToFile writes the container's OCI spec to 'config.json'
+// in the bundle path as expected by the OCI specification.
+func WriteSpecToFile(spec *specs.Spec) error {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	configPath := path.Join(cwd, ociConfigFile)
+	f, err := os.OpenFile(configPath, os.O_RDWR|os.O_CREATE, ociConfigFileMode)
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+	err = json.NewEncoder(f).Encode(&spec)
+	return err
+}
+
+func isValidHook(file os.FileInfo) bool {
```

Comment:
This is a nice clean function. However, I think it would be worth considering "polluting it" a bit with some log calls to help debug hook setup issues (basically a log call for every failure scenario).